bin/cmssw_enable_custom_timeout.py

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+
+""":"
+
+python_cmd="python"
+python3 -c "from FWCore.PythonFramework.CmsRun import CmsRun" 2>/dev/null && python_cmd="python3"
+exec ${python_cmd} $0 ${1+"$@"}
+
+"""
+
+import sys, os
+sys.path.insert(0, os.path.join( os.path.dirname(os.path.abspath(__file__)), '..', 'python'))
+
+import FWCore.ParameterSet.Config as cms
+import pickle
+try: 
+   import argparse
+except ImportError:  #get it from this package instead
+   import archived_argparse as argparse 
+import re
+
+from tweak_program_helpers import make_parser, do_loop, get_cmssw_version, isCMSSWSupported
+
+
+
+def custom_timeout(process, args, timeout):
+
+   process.add_(cms.Service("SiteLocalConfigService",
+                            overrideSourceTimeout = cms.untracked.uint32(timeout))) # XrdAdaptor uses 3*60 unless non-zero value is specified here
```

Comment:
I realized now that a more general way would be to check first if the `SiteLocalConfigService` has already been added to the configuration by something else, e.g. along
```suggestion
   # XrdAdaptor uses 3*60 unless non-zero value is specified here
   if hasattr(process, "SiteLocalConfigService"):
      process.SiteLocalConfigService.overrideSourceTimeout = cms.untracked.uint32(timeout)
   else:
      process.add_(cms.Service("SiteLocalConfigService",
                               overrideSourceTimeout = cms.untracked.uint32(timeout))) 
```

Reply:
Thanks for the suggestion @makortel, I have changed the file. I will mark this as resolved